spec/spec_helper.rb

```diff
@@ -0,0 +1,12 @@
+require 'pry'
+
+# rspec seems to complain about this class not having a #cause method; not sure
+# why but this is a quick hacky fix just to get everything passing. it doesn't
+# seem to affect the results of the tests anyway.
```

Comment:
I really can't work out what's going on here... when running in the app itself then the cause is printed perfectly fine (see `e.cause.ascii_tree`) but under rspec it seems to fail. My guess is that it's something to do with `#cause` being added as a method on Ruby's errors which is conflicting with the method in Parslet of the same name, but I can't really work out why that would be the case and after about an hour of failing to work it out did this hack.

It's a bit of a pain when adding new rules because rspec doesn't show you _why_ the rule has failed to match, but the results of the tests (either matching or raising `ParseFailed`) are the same in either case.